pkgdown/deploy.sh

```diff
@@ -1,14 +1,35 @@
+
+# failure is a natural part of life
 set -e
 
 [ -z "${GITHUB_PAT}" ] && exit 0
+
+echo "[INFO] configuring git"
 git config --global user.email "jaylamb20@gmail.com"
 git config --global user.name "jameslamb"
 
-git clone -q -b gh-pages https://${GITHUB_PAT}@github.com/jameslamb/rsixygen.git output
-cd output
-git rm -rf *
-  cp -r ../docs/* ./
-  git add --all *
-  git commit -m"Update the site from appveyor"
-# echo add files to gh-pages
-git push -q origin gh-pages
+echo "[INFO] creating temporary directory to build repo"
+export OUTPUT_DIR=$(pwd)/output
+mkdir -p ${OUTPUT_DIR}
+
+echo "[INFO] cloning gh-pages branch of the repo"
+git clone \
+    -b gh-pages \
+    https://${GITHUB_PAT}@github.com/jameslamb/rsixygen.git \
+    ${OUTPUT_DIR}
+echo "[INFO] cloned repo to '${OUTPUT_DIR}'"
+
+cd ${OUTPUT_DIR}
+
+echo "[INFO] copying docs to output dir"
+cp -r ../docs/* ./
+
+echo "[INFO] checking files into the repo"
+git add --all *
+
+echo "[INFO]"
+git commit -m "Update the site from appveyor" || exit 0
```

Comment:
hey @BruceZhaoR I want to make you aware of something surprising that I learned tonight. This script was silently causing failures that I couldn't figure out, until I learned this:

If you run `git commit` on a repo with no staged changes, it will return an exit code of `1` (which causes CI to fail).

try it yourself:

```
git commit -m "something"
echo $?
```

So the way this script is set up, PRs like this one that make changes that don't change the `pkgdown` site at all will run into this and get a build failure.

Thank you SO MUCH for contributing this awesome `gh-pages` setup! I'm not tagging you to give you a hard time, just to let you know about this weird thing I found, in case you use a similar script in other projects.